### PR TITLE
Pagination update on page prop change

### PR DESCRIPTION
**Pull request by bojan88**
#### What does this PR do?
Update pagination page state on page prop change. This will allow users to update the page from the parent component.

#### What testing has been done on this PR?
Tested manually and added 1 unit test.

#### How should this be manually tested?
Maybe editing one of the stories to verify this.

#### Any background context you want to provide?
I tested pagination components in 3 other UI frameworks.
They are updating the page on prop change. Also, they are not calling `onChange` on prop change.
One of the drawbacks for this change is an excess component re-render on click. I think we can leave it as is since it's not impacting performance a lot, but saving bytes with simple logic.

#### What are the relevant issues?
I didn't find any relevant issues.

#### Should this PR be mentioned in the release notes?
Maybe as a bug fix

### CODE REVIEW

**Comment by halocline** [comment]
👀

**Review by halocline — APPROVED** [review]
Looks good. Thank you for your contribution.

File: src/js/components/Pagination/__tests__/Pagination-test.js [hunk 1]
```diff
@@ -345,4 +345,25 @@ describe('Pagination', () => {
...
   });
+
+  test(`should change the page on prop change`, () => {
+    const { container, rerender, getByText } = render(
+      <Grommet>
+        <Pagination numberItems={NUM_ITEMS} page={1} />
+      </Grommet>,
+    );
+
+    expect(getByText('1').className).not.toBe(getByText('3').className);
```
Comment (taysea) [inline src/js/components/Pagination/__tests__/Pagination-test.js:356]: Just a small suggestion to align with the convention of other tests in this file. We typically don't reference `className` In this case, maybe you could do something like the test above:
```
const desiredPage = '1';
const activePage = container.querySelector(`[aria-current="page"]`)
      .innerHTML;
expect(activePage).toEqual(desiredPage);

File: src/js/components/Pagination/Pagination.js [hunk 2]
```diff
@@ -177,6 +177,10 @@ const Pagination = forwardRef(
       ...navProps[control],
     }));
 
+    useEffect(() => {
```
Comment (taysea) [inline src/js/components/Pagination/Pagination.js:180]: For readability, I'm wondering if we should move this up to line 44 so it is next to where `activePage` is created.

**Review by taysea — COMMENTED** [review]
Just a couple cleanup comments but functionality seems to be working well. I created an example where a Button outside of pagination component was incrementing the page state value.

File: src/js/components/Pagination/__tests__/Pagination-test.js (hunk 1 above)
Comment (bojan88) [reply src/js/components/Pagination/__tests__/Pagination-test.js]: Thanks. I made it a bit more concise, hope you don't mind. Let me know please if you think I should make it exactly as posted above.

File: src/js/components/Pagination/Pagination.js (hunk 2 above)
Comment (bojan88) [reply src/js/components/Pagination/Pagination.js]: My reasoning to move them at the end was to easily extend it latter if needed, and add dependencies declared above.
But I agree it makes more sense to have ti close to `activePage`

**Review by taysea — APPROVED** [review]
Nice, thanks for incorporating the feedback!

**Comment by ericsoderberghp** [comment]
Thanks for your contribution!

**Comment by mfferreira** [comment]
Thanks for this! The `page` prop value update not updating the active page was driving me nuts already.

**Comment by bojan88** [comment]
@mfferreira I don't quite understand your comment. `page` prop should be updating internal `activePage` in `Pagination` component. Maybe you should open an issue with a better explanation. I'll be happy to work on that, just mention me in the issue comments.

**Comment by B-Esmaili** [comment]
When this change should be released?

**Comment by ShimiSun** [comment]
I'm thinking the upcoming release will be in the next couple of weeks. 
@taysea, @halocline  Is there a workaround we can provide? Otherwise, we may want to re-prioritize the upcoming release.

**Comment by B-Esmaili** [comment]
@halocline What I want to do is changing the page from a higher-order-component. I have already filed an [issue ](https://github.com/grommet/grommet/issues/5232)regarding the issue and I have included a link to code-sandbox link demonstrating the issue.

**Comment by taysea** [comment]
@B-Esmaili I believe Pagination is working as expected. I have tested locally in grommet storybook with the following setup:
```
const [page, setPage] = useState(1);
return (
<>
<Pagination numberItems={237} page={page} />
          <Button
            label="Increment Pagination"
            onClick={() => setPage(page + 1)}
          />
</>
```

The external Button is controlling the value of the pagination. Notice as I click, the active page increases and the internal state of pagination is changing as well:

https://user-images.githubusercontent.com/12522275/117837968-bfbce700-b22e-11eb-9ada-d58c1e438fdc.mov


https://user-images.githubusercontent.com/12522275/117837984-c2b7d780-b22e-11eb-8642-eff91335ef2b.mov

Can you explain how this is not meeting your use case?

**Comment by B-Esmaili** [comment]
@taysea thanks for concerning the issue, did you see my [sandbox](https://codesandbox.io/s/bold-flower-6ic42)? And yes it is working now in the latest version, but did you test it in v2.17.1?

**Comment by taysea** [comment]
> @taysea thanks for concerning the issue, did you see my [sandbox](https://codesandbox.io/s/bold-flower-6ic42)? And yes it is working now in the latest version, but did you test it in v2.17.1?

This fix is currently only on our stable branch, so it will be available in the next release. If you'd like to point your project stable you can try out the fix with these steps: https://github.com/grommet/grommet/wiki/What-is-grommet-stable-and-how-to-use-it%3F

**Comment by B-Esmaili** [comment]
@taysea Thanks for the advice, I'll do it accordingly.

**Pull request merged**